drivers/frequency/adf4368/adf4368.c

```diff
@@ -0,0 +1,1953 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES, INC. “AS IS” AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL ANALOG DEVICES, INC. BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include "adf4368.h"
+#include "no_os_alloc.h"
+#include "no_os_delay.h"
+#include "no_os_error.h"
+#include "no_os_print_log.h"
+
+/* Charge pump current values expressed in uA */
+static const int adf4368_ci_ua[] = {
+	790,
+	990,
+	1190,
+	1380,
+	1590,
+	1980,
+	2390,
+	2790,
+	3180,
+	3970,
+	4770,
+	5570,
+	6330,
+	7910,
+	9510,
+	11100
+};
+
+/**
+ * @brief Writes data to ADF4368 over SPI.
+ * @param dev	   - The device structure.
+ * @param reg_addr - The register address.
+ * @param data 	   - Data value to write.
+ * @return 	   - 0 in case of success or negative error code otherwise.
+ */
+int adf4368_spi_write(struct adf4368_dev *dev, uint16_t reg_addr, uint8_t data)
+{
+	uint8_t buff[ADF4368_BUFF_SIZE_BYTES];
+	uint16_t cmd;
+
+	if (!dev)
+		return -EINVAL;
+
+	cmd = ADF4368_SPI_WRITE_CMD | reg_addr;
+	if (dev->spi_desc->bit_order) {
+		buff[0] = no_os_bit_swap_constant_8(cmd & 0xFF);
+		buff[1] = no_os_bit_swap_constant_8(cmd >> 8);
+		buff[2] = no_os_bit_swap_constant_8(data);
+	} else {
+		buff[0] = cmd >> 8;
+		buff[1] = cmd & 0xFF;
+		buff[2] = data;
+	}
+
+	return no_os_spi_write_and_read(dev->spi_desc, buff,
+					ADF4368_BUFF_SIZE_BYTES);
+}
+
+
+/**
+ * @brief Reads data from ADF4368 over SPI.
+ * @param dev 	   - The device structure.
+ * @param reg_addr - The register address.
+ * @param data	   - Data read from the device.
+ * @return	   - 0 in case of success or negative error code otherwise.
+ */
+int adf4368_spi_read(struct adf4368_dev *dev, uint16_t reg_addr, uint8_t *data)
+{
+	uint8_t buff[ADF4368_BUFF_SIZE_BYTES];
+	uint16_t cmd;
+	int ret;
+
+	if (!dev)
+		return -EINVAL;
+
+	cmd = ADF4368_SPI_READ_CMD | reg_addr;
+	if (dev->spi_desc->bit_order) {
+		buff[0] = no_os_bit_swap_constant_8(cmd & 0xFF);
+		buff[1] = no_os_bit_swap_constant_8(cmd >> 8);
+		buff[2] = no_os_bit_swap_constant_8(ADF4368_SPI_DUMMY_DATA);
+	} else {
+		buff[0] = cmd >> 8;
+		buff[1] = cmd & 0xFF;
+		buff[2] = ADF4368_SPI_DUMMY_DATA;
+	}
+
+	ret = no_os_spi_write_and_read(dev->spi_desc, buff,
+				       ADF4368_BUFF_SIZE_BYTES);
+	if (ret)
+		return ret;
+
+	if (dev->spi_desc->bit_order) {
```

Comment:
drop brackets.

Reply:
Bracket after "if" is forced with /ci/astyle.sh

Reply:
Fixed